201-web-app-loganalytics/azuredeploy.json

```diff
@@ -0,0 +1,180 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "skuName": {
+      "type": "string",
+      "metadata": {
+        "description": "Which Pricing tier our App Service Plan to"
+      },
+      "defaultValue": "S1"
+    },
+    "skuCapacity": {
+      "type": "int",
+      "metadata": {
+        "description": "How many instances of our app service will be scaled out to"
+      },
+      "defaultValue": 1
+
+    },
+    "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "Location for all resources."
+      }
+    },
+    "appName": {
+      "type": "string",
+      "defaultValue": "[uniqueString(resourceGroup().id)]"
+    }
+  },
+  "variables": {
+    "appServicePlanName": "[toLower(concat('asp-', parameters('appName')))]",
+    "webSiteName": "[toLower(concat('wapp-', parameters('appName')))]",
+    "appInsightName": "[toLower(concat('appi-',parameters('appName')))]",
+    "appInisghtsReference": "[concat('microsoft.insights/components/',variables('appInsightName'))]",
+    "logAnalyticsName": "[toLower(concat('la-',parameters('appName')))]"
+  },
+  "resources": [
+    {
+      "apiVersion": "2019-08-01",
+      "name": "[variables('appServicePlanName')]",
+      "type": "Microsoft.Web/serverfarms",
+      "location": "[parameters('location')]",
+      "tags": {
+        "displayName": "HostingPlan",
+        "ProjectName": "[parameters('appName')]"
+      },
+      "sku": {
+        "name": "[parameters('skuName')]",
+        "capacity": "[parameters('skuCapacity')]"
+      },
+      "properties": {
+        "name": "[variables('appServicePlanName')]"
+      }
+    },
+
+    {
+      "apiVersion": "2019-08-01",
+      "name": "[variables('webSiteName')]",
+      "type": "Microsoft.Web/sites",
+      "location": "[parameters('location')]",
+      "tags": {
+        "displayName": "Website",
+        "ProjectName": "[parameters('appName')]"
+      },
+      "dependsOn": [
+        "[resourceId('Microsoft.Web/serverfarms/', variables('appServicePlanName'))]",
+        "[resourceId('Microsoft.OperationalInsights/workspaces',variables('logAnalyticsName'))]"
+      ],
+      "identity": {
+        "type": "SystemAssigned"
+      },
+      "properties": {
+        "serverFarmId": "[resourceId('Microsoft.Web/serverfarms', variables('appServicePlanName'))]",
+        "httpsOnly": true,
+        "siteConfig": {
+          "minTlsVersion": "1.2"
+        }
+      },
+      "resources": [
+        {
+          "apiVersion": "2019-08-01",
+          "name": "appsettings",
+          "type": "config",
+          "dependsOn": [
+            "[resourceId('Microsoft.Web/Sites', variables('WebsiteName'))]",
+            "Microsoft.ApplicationInsights.AzureWebSites",
+            "[resourceId('microsoft.insights/components/', variables('appInsightName'))]"
```

Comment:
you have a variable defined for this - so let's be consistent and use it everywhere or use the resourceId function everywhere - either is fine, but not both.